LoopKitUI/View Controllers/InsulinSensitivityScheduleViewController.swift

```diff
@@ -0,0 +1,397 @@
+        tableView.register(TextButtonTableViewCell.self, forCellReuseIdentifier: TextButtonTableViewCell.className)
+        updateEditButton()
+    }
+
+    @objc private func cancel(_ sender: Any?) {
+        self.navigationController?.popViewController(animated: true)
+    }
+
+    // MARK: - State
+
+    public var insulinSensitivityScheduleStorageDelegate: InsulinSensitivityScheduleStorageDelegate?
+
+    public var unit: HKUnit = HKUnit.milligramsPerDeciliter.unitDivided(by: .internationalUnit())
+
+    public var schedule: InsulinSensitivitySchedule? {
+        get {
+            let validEntries = internalItems.compactMap { (item) -> RepeatingScheduleValue<Double>? in
+                guard let value = item.value else {
+                    return nil
+                }
+                return RepeatingScheduleValue(startTime: item.startTime, value: value)
+            }
+            return InsulinSensitivitySchedule(unit: unit, dailyItems: validEntries, timeZone: timeZone)
+        }
+        set {
+            if let newValue = newValue {
+                unit = newValue.unit
+                internalItems = newValue.items.map { (entry) -> RepeatingScheduleValue<Double?> in
+                    RepeatingScheduleValue(startTime: entry.startTime, value: entry.value)
+                }
+                isScheduleModified = false
+            }
+        }
+    }
+
+    private var internalItems: [RepeatingScheduleValue<Double?>] = [] {
+        didSet {
+            isScheduleModified = true
+            updateInsertButton()
+        }
+    }
+
+    let allowedValues: [Double]
+    let minimumTimeInterval: TimeInterval
+
+    var lastValidStartTime: TimeInterval {
+        return TimeInterval.hours(24) - minimumTimeInterval
+    }
+
+    private var isScheduleModified = false {
+        didSet {
+            if isScheduleModified {
+                self.navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: .cancel, target: self, action: #selector(cancel(_:)))
+            } else {
+                self.navigationItem.leftBarButtonItem = nil
+            }
+            updateSyncButton()
+        }
+    }
+
+    private func isValid(_ value: Double?) -> Bool {
+        guard let value = value else {
+            return false
+        }
+        return allowedValues.contains(value)
+    }
+
+    public var isScheduleValid: Bool {
+        return !internalItems.isEmpty &&
+            internalItems.allSatisfy { isValid($0.value) }
+    }
+
+
+    private func updateSyncButton() {
+        if let cell = tableView.cellForRow(at: IndexPath(row: 0, section: Section.sync.rawValue)) as? TextButtonTableViewCell {
+            cell.isEnabled = isScheduleModified && isScheduleValid
+        }
+    }
+
+    private func updateEditButton() {
+        editButtonItem.isEnabled = internalItems.endIndex > 1
```

Comment:
Also, `!isEditing &&`???

Reply:
The button changes to "Done" when editing, so we don't want to disable that.